File: omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/pod/command/insulin/program/BolusShortInsulinProgramElement.java
package info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.command.insulin.program;

import java.nio.ByteBuffer;

public class BolusShortInsulinProgramElement implements ShortInsulinProgramElement {
    private short numberOfPulses;

    public BolusShortInsulinProgramElement(short numberOfPulses) {
        this.numberOfPulses = numberOfPulses;
    }

    public short getNumberOfPulses() {
        return numberOfPulses;
    }

    @Override public byte[] getEncoded() {
        return ByteBuffer.allocate(2).putShort(numberOfPulses).array();
    }
}

File: pump/omnipod/dash/src/main/kotlin/app/aaps/pump/omnipod/dash/driver/pod/command/insulin/program/BolusShortInsulinProgramElement.kt
package app.aaps.pump.omnipod.dash.driver.pod.command.insulin.program

import java.nio.ByteBuffer

class BolusShortInsulinProgramElement(
    private val numberOfPulses: Short
) : ShortInsulinProgramElement {

    override val encoded: ByteArray
        get() = ByteBuffer.allocate(2).putShort(numberOfPulses).array()
}
